packages/check/match.js

```diff
@@ -272,6 +275,17 @@ const testSubtree = (value, pattern) => {
   if (pattern === Object) {
     pattern = Match.ObjectIncluding({});
   }
+  // This must be invoked before pattern instanceof Array as strings are regarded as arrays
+  // We invoke the pattern as IIFE so that `pattern isntanceof Where` catches it 
+  if (pattern === Match.NonEmptyString) {
+    pattern = (function() {
+      const condition = (val) => {
+         check(val, String);
+         return val.length > 0;
+      }
+      return new Where(condition);
+    })();
+  }
```

Comment:
1. That's exactly what `Symbol`s are for, though I'm not sure whether it wouldn't increase minimum required browser version.
2. It'd be best not to create this helper function and `Where` instance every time, but create it on a top-level and only use here.

Reply:
The new code still creates new closure and `Where` instance every time.

Reply:
How do you propose to fix it?

Reply:
```ts
function nonEmptyStringCondition(value) {
  check(value, String);
  return value.length > 0;
}

const NonEmptyString = new Where(nonEmptyStringCondition);
```

Reply:
How do I put it 😅 ...
The last version you pushed reuses the `nonEmptyStringCondition`, sure. But it creates a new `Where` instance each time. I proposed to reuse the `Where` instance too.

Reply:
sawry tfw no english